### PR TITLE
[add] topUserList

### LINKED ISSUE #115: 活跃用户列表
## 后端接口

https://hackathon-api.kaiyuanshe.cn/swagger/#operations-OpenHackathon-get_v2_user_topUsers

image

## 界面位置

image

### 组件

https://github.com/kaiyuanshe/OpenHackathon-Web/blob/main/components/User/UserList.tsx

### 模型

https://github.com/kaiyuanshe/OpenHackathon-Web/blob/ba01208baeb0d0fdc2a4c90c2f97084ef7a2d59f/models/User.ts#L114-L136

### PR DESCRIPTION
**Pull request by Five-great**
- Closes #115 
- Rebuild #117

### CODE REVIEW

**Comment by Five-great** [comment]
浏览链接 https://openhackathonweb.vercel.app/

File: components/User/TopUserList.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    <Col sm={12} xs={12} lg={7}>
+      <div className={`shadow rounded-3 ${styles.showTitle}`}>
+        <div className={styles.showMedal}>
+          <i className="d-inline-block ms-1 overflow-hidden rounded-circle" />
+          <i className="d-inline-block ms-1 overflow-hidden rounded-circle" />
+          <i className="d-inline-block ms-1 overflow-hidden rounded-circle" />
+        </div>
+        <h3
+          data-text="黑客馆"
+          className="position-relative m-auto mb-0  fw-bold text-center"
```
Comment (TechQuery) [inline components/User/TopUserList.tsx:23]:
```suggestion
          className="position-relative m-auto mb-0 fw-bold text-center"
```

File: components/User/TopUserList.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        as="ul"
+        className={`mt-2 g-0 align-items-end text-center ps-0 pt-2 list-unstyled ${styles.topUserUl}`}
+      >
+        {value.slice(0, 3).map(({ userId, score, user }) => (
+          <Col as="li" key={userId}>
+            <div
+              className={`shadow-lg overflow-hidden rounded-circle m-auto ${styles.imgBox}`}
+            >
+              <Image
+                className={'w-100'}
```
Comment (TechQuery) [inline components/User/TopUserList.tsx:38]:
```suggestion
                className="w-100"
```

File: components/User/TopUserList.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                src={user?.photo}
+                alt={
+                  user?.nickname || user?.name || user?.username || '神秘黑客'
+                }
+              />
+            </div>
+            <div
+              className={`position-relative overflow-hidden  ${styles.showBox} `}
```
Comment (TechQuery) [inline components/User/TopUserList.tsx:46]:
```suggestion
              className={`position-relative overflow-hidden ${styles.showBox}`}
```

File: pages/index.tsx [hunk 4]
```diff
@@ -62,6 +66,18 @@ const HomePage = ({
       </Button>
     </section>
 
+    <div
+      className="my-5 text-center"
+      style={{
+        background: 'linear-gradient(#F8F9FA,#fff)',
+        marginTop: '-3rem!important',
```
Comment (TechQuery) [inline pages/index.tsx:73]:
```suggestion
        marginTop: '-3rem',
```

File: pnpm-lock.yaml [hunk 5]
```diff
@@ -8370,4 +8370,4 @@ packages:
     resolution: {integrity: sha512-rOS09Z2HSVGFs2dn/TuYk5BlCaZcVe8UDLLjj1ySYF828LATKKdxuakSZMvrDz54yiKPDYVfjdKqcX8Jky3BIA==}
     dependencies:
       tslib: 2.3.0
-    dev: false
+    dev: false
```
Comment (TechQuery) [inline pnpm-lock.yaml:8373]:
```suggestion
    dev: false

```

File: styles/TopUserList.module.less [hunk 6]
```diff
@@ -0,0 +1,195 @@
+@logo-img: 'https://hackathon-api.static.kaiyuanshe.cn/6342619375fa1817e0f56ce1/2022/10/09/logo22.jpg';
+.topUserRow {
```
Comment (TechQuery) [inline styles/TopUserList.module.less:2]:
```suggestion
@logo-img: 'https://hackathon-api.static.kaiyuanshe.cn/6342619375fa1817e0f56ce1/2022/10/09/logo22.jpg';

.topUserRow {
```

File: styles/TopUserList.module.less [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+  }
+  50% {
+    position: relative;
+    top: -0.8rem;
+  }
+  100% {
+    position: relative;
+    top: 0px;
+  }
+} ;
```
Comment (TechQuery) [inline styles/TopUserList.module.less:195]:
```suggestion
}
```

**Review by TechQuery — CHANGES_REQUESTED** [review]

**Review by TechQuery — APPROVED** [review]

**Pull request merged**